test-dts/mount.d-test.ts

```diff
@@ -48,10 +52,12 @@ wrapper = mount(AppWithProps, {
 // vm is properly typed
 expectType<string>(wrapper.vm.a)
 
-// can receive extra props
-mount(AppWithProps, {
-  props: { a: 'Hello', b: 2 }
-})
+// can't receive extra props
+expectError(
```

Comment:
Would this error out for transparent components without `props` defined?

Reply:
Can you give me an example of what you mean?

Reply:
If a component has no props defined, and passes them down via `$attrs`.

```js
const Component = {
   template: `<div><ChildComponent v-bind="$attrs"/></div>`
}
const wrapper = mount(Component, { props: { a:'a' } })

expect(wrapper.findComponent(ChildComponent).props('a')).toBe('a')
```

Reply:
Ok, so yes: I expect this to error out if the props are not defined (as they end up with the `never` type). I think it makes sense to error out, as it is not the most common case, and let the developer use `mount(Component, { props: { a:'a' } as any })` in that case. But I'm open to your suggestions if you think this is too constraint.

Reply:
This wont error out for plain JS use right? 

Erroring out is technically wrong, as the Vue API allows you to pass props to that kind of components, but as long as we state the reasoning behind this, should be fine :)

You are right, its not a common use case and having type safety for the 80% of people is better, we just need to make sure these little cases have workarounds for the more novice TS users out there.

Reply:
Yes this is for TS developers only, JS will be fine, sorry if that wasn't clear :)

And even in TS it still works, it just forces the developers to cast the `props` if they want to use it with undeclared props.

IMO, TS developers will be happy to have intellisense/autocomplete/type-checking in tests, and won't be too lost if the compilation indicates that the props does not match the props of the component. But you're 100% right that we need to mention it the docs 👍